### DISCUSSION TITLE
What is WelsonJS for?

### DISCUSSION THREAD

**Discussion by squadjot** [gnh1201/welsonjs#161] [General]
I'm a long time user/creator of HTA applications.
I'm not fully understanding what you're are accomplishing with WelsonJS.

It look's interesting.. but also very complex ..and feels a bit bloated. 
But, maybe i just don't understand the purpose of it all.  ?

I assume we still end up with presenting HTML in a legacy IEBrowser, with a legacy JS engine, right?

What is the prime example of what you can do in WelsonJS, but not in a HTA?

**Comment by gnh1201** [comment]
As mentioned in the [***Introduction***](https://catswords-oss.rdbl.io/5719744820/5330609327), WelsonJS was initiated to support an environment where all tasks must be handled exclusively using built-in components. Thus, most tasks can be performed with JavaScript, while users with fewer restrictions can utilize auxiliary components written in C#.

**Comment by gnh1201** [comment]
As highlighted in [SECURITY.MD](https://github.com/gnh1201/welsonjs/blob/master/SECURITY.MD), while there are numerous use cases, the most profitable field has been [SERP](https://catswords-oss.rdbl.io/5719744820/1706431912).

**Comment by lukasMega** [comment]
is possible to create single-file executable (.exe) app?
Or it's limited to use `WelsonJS.Launcher` ?

**Reply by squadjot** [reply]
> is possible to create single-file executable (.exe) app? Or it's limited to use `WelsonJS.Launcher` ?

I'm not sure this will pack everything into one file, but you can at least convert the HTA to EXE.
https://github.com/jeremyben/thebatchman

**Reply by gnh1201** [reply]
It’s not impossible. Theoretically, you could embed a ZIP file containing all the project contents into the internal resources (e.g., `resources.resx`) of the `WelsonJS.Launcher` program. There are also several other methods. I’m currently considering the optimal approach.
